ui/admin-portal/src/app/components/tasks/assign-tasks-list/assign-tasks-list.component.ts

```diff
@@ -117,31 +117,28 @@ export class AssignTasksListComponent implements OnInit {
     if(task.name === 'duolingoTest') {
```

Comment:
Use the button passing a list to avoid all this special "if task.name === duolingoTest" stuff

The big change here is that assigning a coupon automatically creates the task - rather than having a special task that also assigns the coupon. The task for them is to use the coupon. 

If a user is given a task to use a coupon and they haven't been assigned one, they can report back that they can't use a coupon because they haven't been assigned one.